src/core/commandVersionUpgrades/canonicalizeAndValidateCommand.ts

```diff
@@ -128,11 +126,9 @@ export async function checkForOldInference(
   });
 
   if (hasOldInference) {
-    const hideInferenceWarning =
-      graph.extensionContext.globalState.get<boolean>(
-        globalStateKeys.hideInferenceWarning,
-        false
-      );
+    const hideInferenceWarning = graph.ide.globalState.get(
+      "hideInferenceWarning"
+    );
```

Comment:
I improved the type inference a bit, so you don't need to put the `<boolean>` anymore.  I think it's now pretty slick

Reply:
Oh that is neat!